fe/fe-core/src/main/java/com/starrocks/connector/RemoteFileOperations.java

```diff
@@ -136,6 +139,80 @@ public List<RemoteFileInfo> getRemoteFiles(List<Partition> partitions, Options o
         return resultRemoteFiles;
     }
 
+    public RemoteFileInfoSource getRemoteFilesAsync(Table table, List<Partition> partitions, GetRemoteFilesParams params) {
+        RemoteFileScanContext scanContext = new RemoteFileScanContext(table);
+
+        // compute cache miss size.
+        {
+            List<RemotePathKey> remotePathKeys = new ArrayList<>();
+            for (Partition partition : partitions) {
+                RemotePathKey key = RemotePathKey.of(partition.getFullPath(), isRecursive);
+                remotePathKeys.add(key);
+            }
+
+            int cacheMissSize = partitions.size();
+            if (enableCatalogLevelCache && params.isUseCache()) {
+                cacheMissSize = cacheMissSize - remoteFileIO.getPresentRemoteFiles(remotePathKeys).size();
+            }
+            Tracers.count(Tracers.Module.EXTERNAL, HMS_PARTITIONS_REMOTE_FILES, cacheMissSize);
+        }
+
+        class MyAsyncTaskQueue extends AsyncTaskQueue<RemoteFileInfo> implements RemoteFileInfoSource {
```

Comment:
why named `MyAsyncTaskQueue`?

Reply:
it's just a internal class name. no big deal.